be/src/exprs/arithmetic_operation.h

```diff
@@ -182,6 +191,61 @@ struct ArithmeticBinaryOperator {
             static_assert(is_binary_op<Op>, "Invalid binary operators");
         }
     }
+
+    template <typename ResultType>
+    static llvm::Value* generate_ir(llvm::IRBuilder<>& b, llvm::Value* l, llvm::Value* r) {
+        return generate_ir<ResultType, ResultType, ResultType>(b, l, r);
+    }
+
+    template <typename LType, typename RType, typename ResultType>
+    static llvm::Value* generate_ir(llvm::IRBuilder<>& b, llvm::Value* l, llvm::Value* r) {
+        if constexpr (is_add_op<Op>) {
+            if constexpr (lt_is_float<Type>) {
+                return b.CreateFAdd(l, r);
+            } else {
+                return b.CreateAdd(l, r);
+            }
+        } else if constexpr (is_sub_op<Op>) {
+            if constexpr (lt_is_float<Type>) {
+                return b.CreateFSub(l, r);
+            } else {
+                return b.CreateSub(l, r);
+            }
+        } else if constexpr (is_mul_op<Op>) {
+            // TODO(Yueyang): implement float type * 0.
+            if constexpr (lt_is_float<Type>) {
+                return b.CreateFMul(l, r);
+            } else {
+                return b.CreateMul(l, r);
+            }
```

Comment:
do they keep the same behaviors like non-scalar code? for example,  (int_8) 129 + (int_8) 129 = ?